### PR TITLE
Add tests and refactor pull-request-page

### PR DESCRIPTION
**Pull request by txsmith**
Wherever possible, the pull request object is passed to child elements instead of the array of all PR's along with the currently selected index.

### CODE REVIEW

File: test/projects/pull-request/pull-request-page.html [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+        base: { label: 'preview-app:master' },
+        created_at: "2016-05-25T00:31:28Z",
+        assignee: { html_url: 'https://example.com/',
+                    login: 'reviewer' }
+      };
+
+      var elem;
+
+      setup(function() {
+        elem = fixture('basic');
```
Comment (eanker) [inline test/projects/pull-request/pull-request-page.html]: Shouldn't it be better to do some better naming?
elem and basic are arbitrary names.
The rest of the tests have some more descriptive names.

File: src/projects/projects-page.html [hunk 2]
```diff
@@ -123,15 +122,15 @@
         activeProject: Object
       },
       observers: [
-        'chooseProject(projectTitle.*)'
+        'chooseProject(projectTitle.*, pr.*)'
```
Comment (TimvdLippe) [inline src/projects/projects-page.html]: Why was this required?

**Comment by TimvdLippe** [comment]
LGTM!
